crates/biome_js_analyze/src/semantic_analyzers/nursery/no_this_in_static.rs

```diff
@@ -0,0 +1,127 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic, ActionCategory};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::{JsMethodClassMember, JsStaticMemberExpression, JsSuperExpression, JsThisExpression, JsClassDeclaration};
+use biome_rowan::{declare_node_union, AstNode, AstNodeList, AstNodeExt, BatchMutationExt};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Succinct description of the rule.
+    ///
+    /// Put context and details about the rule.
+    /// As a starting point, you can take the description of the corresponding _ESLint_ rule (if any).
+    ///
+    /// Try to stay consistent with the descriptions of implemented rules.
+    ///
+    /// Add a link to the corresponding ESLint rule (if any):
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/rule-name
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = 1;
+    /// a = 2;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// ```
+    ///
+    pub(crate) NoThisInStatic {
+        version: "next",
+        name: "noThisInStatic",
+        recommended: false,
+    }
+}
+
+declare_node_union! {
+    pub(crate) JsThisSuperExpression = JsSuperExpression | JsThisExpression
+}
+
+impl Rule for NoThisInStatic {
+    type Query = Ast<JsThisSuperExpression>;
+    type State = JsStaticMemberExpression;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let this_super_expression = ctx.query();
+
+        let static_method = this_super_expression
+            .syntax()
+            .ancestors()
+            .find_map(JsMethodClassMember::cast)
+            .filter(|member| {
+                member
+                    .modifiers()
+                    .iter()
+                    .any(|modifier| modifier.as_js_static_modifier().is_some())
+            });
+            
+            if  static_method.is_some() {
+                    this_super_expression
+                        .syntax()
+                        .ancestors()
+                        .find_map(JsStaticMemberExpression::cast)
```

Comment:
I think we could move this code in `diagnostics. We could simply return `Some(())`.

The state could be defined as follows:

```
    type State = ();
```

Reply:
After adding this change it I dont know why it is only squigglying the last instance of `this`/`super` it finds, seems to me it is running diagnostics only once (?)

Reply:
This is solved, I was mishandling the `Option` values